### PR TITLE
🏞️ Rovin' Kovan

### PR DESCRIPTION
**Pull request by mhluongo**
Moving Kovan testnet support to be accessible via the network selector. This implies making a ton of code multi-network aware, though still short of production support as in #5 or #7.

- [x] Support multiple networks in `ChainService` 
- [x] Support multiple networks in `IndexingService`
- [x] Support multiple networks in `KeyringService` 
- [x] Kovan-specific gas price estimates
- [x] Allow new networks to be added for tracking via Redux
- [ ] Track down and nuke all instances of `getEthereumNetwork()` in the background script
- [x] Connect the network selector to Redux
  - [x] Track all accounts by `AddressNetwork` rather than just address
  - [ ] Switch the dApp provider network (#733)

What this PR *won't* do....
* Add all accounts to track on new network whenever network is changed. This was originally a goal, but in retrospect we only want to track accounts that we've seen the user visit on a particular network to save resources
* Include any sort of multi-network asset or asset transfer/historical transaction support.
* Update token lists to track tokens on Kovan
* Pull prices for assets on other networks
* #776

The user-facing bits here are feature-flagged.

### CODE REVIEW

**Comment by Shadowfiend** [comment]
> Commit messages are the piece of knowledge that is second most likely to survive tool transitions (the first is the code itself); as such, they must stand alone. Do not reference tickets or issues in your commit messages. Summarize any conclusions from the issue or ticket that inform the commit itself, and capture any additional reasoning or context in the merge commit. See 
Merging PRs
 for more.

**Comment by mhluongo** [comment]
Draft commit history :stuck_out_tongue:

File: background/lib/gas.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          confidence: 95,
+          maxPriorityFeePerGas: 1500000000n,
+          maxFeePerGas: (baseFeePerGas * 15n) / 1n + 1500000000n,
+          price: (gasPrice * 9n) / 10n,
+        },
+        {
+          confidence: 70,
+          maxPriorityFeePerGas: 2500000000n,
+          maxFeePerGas: (baseFeePerGas * 13n) / 1n + 1100000000n,
+          price: (gasPrice * 8n) / 10n,
```
Comment (wojciech-turek) [inline background/lib/gas.ts:71]: @mhluongo Shouldn't that be `/ 10n` instead of 1n?

Comment (mhluongo) [reply background/lib/gas.ts:71]: Lol yep!

File: background/services/indexing/index.ts [hunk 2]
```diff
@@ -290,8 +291,19 @@ export default class IndexingService extends BaseService<Events> {
     addressNetwork: AddressNetwork,
     contractAddresses?: HexString[]
   ): ReturnType<typeof getTokenBalances> {
+    const provider = this.chainService.requirePollingProvider(
+      addressNetwork.network
+    )
+
+    const isAlchemy = (p: any): p is AlchemyProvider =>
+      typeof (provider as AlchemyProvider).apiKey !== "undefined"
```
Comment (Shadowfiend) [inline background/services/indexing/index.ts:304]: Can't we just use `instanceof`?

Comment (mhluongo) [reply background/services/indexing/index.ts:304]: Probably, as long as it covers subclasses

**Comment by mhluongo** [comment]
I think we leave it for a bit longer as draft, then close as we start QA'ing the replacement? It's a nice reminder of the branch to crib off in the meantime.

**Comment by Shadowfiend** [comment]
Closing this now.

**Comment by mhluongo** [comment]
Goodnight sweet prince

**Comment by Shadowfiend** [comment]
💕💐🫡

**Pull request closed without merging**